File: 1455.check-if-a-word-occurs-as-a-prefix-of-any-word-in-a-sentence.cpp
/*
 * @lc app=leetcode id=1455 lang=cpp
 *
 * [1455] Check If a Word Occurs As a Prefix of Any Word in a Sentence
 */
#include <string>
using namespace std;
// @lc code=start
class Solution
{
public:
    int isPrefixOfWord(string sentence, string searchWord)
    {
        int pos = 0;
        int searchWord_width = searchWord.size();
        int index = 1;
        while (pos != -1)
        {
            pos = sentence.find_first_not_of(' ', pos);
            bool match = true;
            for (int i = 0; i != searchWord_width; ++i)
            {
                if (sentence[pos + i] != searchWord[i])
                {
                    match = false;
                    break;
                }
            }
            if (match)
            {
                return index;
            }
            ++index;
            pos = sentence.find_first_of(' ', pos);
        }
        return -1;
    }
};
// @lc code=end

File: 1455.check-if-a-word-occurs-as-a-prefix-of-any-word-in-a-sentence.java
/*
 * @lc app=leetcode id=1455 lang=java
 *
 * [1455] Check If a Word Occurs As a Prefix of Any Word in a Sentence
 */

// @lc code=start
class Solution {
    public int isPrefixOfWord(String sentence, String searchWord) {
        int pos = 0;
        int searchWord_width = searchWord.length();
        int index = 1;
        while (pos != -1) {
            while (!Character.isAlphabetic(sentence.charAt(pos))) {
                ++pos;
            }
            boolean match = true;
            for (int i = 0; i != searchWord_width; ++i) {
                if (sentence.charAt(pos + i) != searchWord.charAt(i)) {
                    match = false;
                    break;
                }
            }
            if (match) {
                return index;
            }
            ++index;
            pos = sentence.indexOf(' ', pos);
        }
        return -1;
    }
}
// @lc code=end
